### ISSUE TITLE
Scrolling to item

### ISSUE THREAD

**Issue by SamiKamal** [catamphetamine/virtual-scroller#21]
Hey, thanks for the amazing work here ❤️ 

I'm utilizing virtual-scroller to display a vast list of items (10,000+). I'm aiming to scroll the user to a particular item based on its index (e.g., navigating to index 5931). Currently, I determine the scroll position by:

1. Obtaining the`getAverageItemHeight`.
2. Multiplying it by the `itemIndex`.
However, this method lacks precision because the`getAverageItemHeight` can easily deviate. Is there a more accurate approach to determine the scroll position?

**Comment by catamphetamine** [comment]
Hi. If the item hasn’t been rendered yet then its top position is unknown.
In those cases, precise scroll-to is impossible.

**Comment by SamiKamal** [comment]
Umm, wouldn't it be possible if the following happen:
1. Implement custom smooth scrolling
2. Add `scrollToItem(index)` function
3. When `scrollToItem` is called scroll to the end by default
4. As we scroll the items will be rendered
5. If we came across the item while scrolling then cancel the scroll

Not sure if this is practical though.

Also one random question, would it be possible to fire an event or call a callback when the initial render is done?

**Comment by catamphetamine** [comment]
> Umm, wouldn't it be possible if the following happen:

I don't think that it would be practical because the scrolling wouldn't be smooth: it would scroll almost to the end, wait for it to render, then scroll almost to the end, wait for it to render, etc, until item N is rendered.

> Also one random question, would it be possible to fire an event or call a callback when the initial render is done?

There's a `onItemInitialRender(item)` option/property. See the readme (expandable sections).

**Comment by sachingooo** [comment]
Love this functionality as well! I agree that it would be **_fantastic_** to be able to programmatically scroll to a given element index
